### DISCUSSION TITLE
Release official musl binaries (including `libduckdb.so`)

### DISCUSSION THREAD

**Discussion by arjunbajaj** [duckdb/duckdb#12523] [Ideas]
Hi,

Consider releasing official `musl` binaries for DuckDB and `libduckdb.so` with all the extensions bundled. It already compiles on Alpine Linux, but extensions like `sqlite_scanner` need to be compiled and loaded separately.

If an official release for `libduckdb` exists, it'll greatly help projects like Elixir's [Explorer/Adbc](https://github.com/elixir-explorer/adbc) which can then use their standard procedure to load and run DuckDB seamlessly on Alpine Linux.

I have documented a [hacky workaround here](https://github.com/elixir-explorer/adbc/issues/92#issuecomment-2166608331) in the meantime, but that doesn't include SQLite and some of the other extensions, but does include `httpfs` and `parquet`.

**Comment by Euraxluo** [comment]
Request to advance this

**Comment by jpalawaga** [comment]
Yes please. Being able to use alpine/remove glibc requirement would be great.

**Reply by szarnyasg** [reply]
@jpalawaga this is on our [roadmap](https://duckdb.org/roadmap#future-work--looking-for-funding), for v1.5 (2026 Q1) or v1.6 (2026 Q2).

In the meantime, you can try building from source: https://duckdb.org/docs/stable/dev/building/linux#alpine-linux

**Comment by derhuerst** [comment]
From the [DuckDB v1.5.0 announcement post](https://duckdb.org/2026/03/09/announcing-duckdb-150#spatial-extension):

> ### [CLI for Linux with musl libc](https://duckdb.org/2026/03/09/announcing-duckdb-150#cli-for-linux-with-musl-libc)
> We are distributing CLI clients that work with [musl libc](https://duckdb.org/docs/stable/dev/building/linux.html) (e.g., for Alpine Linux, commonly used in Docker images). The archives are available [on GitHub](https://github.com/duckdb/duckdb/releases/tag/v1.5.0).
> Note that the musl libc CLI client requires the libstdc++. To install this package, run:
> ```shell
> apk add libstdc++
> ```
